sway/tree/workspace.c

```diff
@@ -233,19 +228,19 @@ static bool _workspace_by_number(struct sway_container *view, void *data) {
 	if (view->type != C_WORKSPACE) {
 		return false;
 	}
-	struct workspace_by_number_data *wbnd = data;
-	int a = strspn(view->name, wbnd->cset);
-	return a == wbnd->len && strncmp(view->name, wbnd->name, a) == 0;
+	char *name = data;
+	char *view_name = view->name;
+	while (isdigit(*name)) {
+		if (*name++ != *view_name++) {
```

Comment:
Could this block of code be replaced with something like this?

    int num = atoi(data);
    int view_num = atoi(view->name);
    return num == view_num && num > 0;

Reply:
-1

Reply:
Why the -1? Doesn't it achieve the same thing with less code?

Reply:
Less code isn't always desirable. I think your suggestion is more complex.